ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Tests/Timeouts.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE LambdaCase     #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+module Test.Consensus.PeerSimulator.Tests.Timeouts (tests) where
+
+import           Cardano.Slotting.Time (SlotLength, slotLengthFromSec)
+import           Control.Monad.IOSim (runSimOrThrow)
+import           Data.List.NonEmpty (NonEmpty ((:|)))
+import qualified Data.Map as Map
+import           Data.Maybe (fromJust)
+import           Data.Time.Clock (diffTimeToPicoseconds)
+import           Ouroboros.Consensus.Block (getHeader)
+import           Ouroboros.Consensus.Util.Condense
+import           Ouroboros.Consensus.Util.IOLike (DiffTime, fromException)
+import qualified Ouroboros.Network.AnchoredFragment as AF
+import           Ouroboros.Network.Block (tipFromHeader)
+import           Ouroboros.Network.Driver.Limits
+                     (ProtocolLimitFailure (ExceededTimeLimit))
+import           Ouroboros.Network.Protocol.ChainSync.Codec (mustReplyTimeout)
+import           Test.Consensus.BlockTree (btTrunk)
+import           Test.Consensus.Genesis.Setup
+import           Test.Consensus.Network.Driver.Limits.Extras (chainSyncTimeouts)
+import           Test.Consensus.PeerSimulator.Run (SchedulerConfig (..))
+import           Test.Consensus.PeerSimulator.StateView
+import           Test.Consensus.PointSchedule
+import qualified Test.QuickCheck as QC
+import           Test.QuickCheck
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+import           Test.Util.Orphans.IOLike ()
+
+tests :: TestTree
+tests = testProperty "timeouts" prop_timeouts
+
+prop_timeouts :: QC.Gen QC.Property
+prop_timeouts = do
+  genesisTest <- genChains 0
+
+  let scChainSyncTimeouts = chainSyncTimeouts slotLength (gtHonestAsc genesisTest)
+      schedulerConfig = SchedulerConfig {scChainSyncTimeouts}
+
+  let schedule =
+        dullSchedule
+          (fromJust $ mustReplyTimeout scChainSyncTimeouts)
+          (btTrunk $ gtBlockTree genesisTest)
+
+  pure $ withMaxSuccess 10 $ runSimOrThrow $
+    runTest schedulerConfig genesisTest schedule $ \stateView ->
+      case svChainSyncExceptions stateView of
+        [] ->
+          counterexample ("result: " ++ condense (svSelectedChain stateView)) False
+        [exn] ->
+          case fromException $ cseException exn of
+            Just (ExceededTimeLimit _) -> counterexample "REVIEW" True
```

Comment:
what's the issue here? shouldn't it just be "timed out"?

Reply:
Fixed in https://github.com/input-output-hk/ouroboros-consensus/pull/452/commits/c09d8d2feb87ccf01916ffe601812221ffae54e5.